units/patch_intent.go

```diff
@@ -535,6 +526,69 @@ func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patc
 	return previousPatch.Status, nil
 }
 
+// updatePatchDocPatches updates the commit sha for existing patch modules and adds new modules to the patch for modules
+// that exist in the project but not in the patch.  This is necessary so that the manifest load endpoint will include the
+// updated commit sha in the response's module overrides field.
+func updatePatchDocPatches(ctx context.Context, patchDoc *patch.Patch, project *model.Project, token string) error {
+	var err error
+	for i := range patchDoc.Patches {
+		if patchDoc.Patches[i].ModuleName != "" {
+			// validate the module exists
+			var module *model.Module
+			module, err = project.GetModuleByName(patchDoc.Patches[i].ModuleName)
+			if err != nil {
+				return errors.Wrapf(err, "finding module '%s'", patchDoc.Patches[i].ModuleName)
+			}
+			if module == nil {
+				return errors.Errorf("module '%s' not found", patchDoc.Patches[i].ModuleName)
+			}
+			branch, err := getBranch(ctx, module.Repo, patchDoc.Patches[i].ModuleName, module.Branch, token)
+			if err != nil {
+				return errors.Wrapf(err, "getting branch for module '%s'", patchDoc.Patches[i].ModuleName)
+			}
+			patchDoc.Patches[i].Githash = *branch.Commit.SHA
+		}
+	}
+	for _, mod := range project.Modules {
+		patchFound := false
+		for _, p := range patchDoc.Patches {
+			if p.ModuleName == mod.Name {
+				patchFound = true
+				break
+			}
+		}
+		if patchFound {
+			continue
+		}
+		branch, err := getBranch(ctx, mod.Repo, mod.Name, mod.Branch, token)
+		if err != nil {
+			return errors.Wrapf(err, "getting branch for module '%s'", mod.Name)
+		}
+		patchDoc.Patches = append(patchDoc.Patches, patch.ModulePatch{
+			Githash:    *branch.Commit.SHA,
+			ModuleName: mod.Name,
+		})
+
+	}
+	return nil
+}
+
+// getBranch retrieves the branch for the given repo, module, and branch name.
+func getBranch(ctx context.Context, repo, name, moduleBranch, token string) (*github.Branch, error) {
```

Comment:
Should this be moved somewhere else (maybe to the manifest file?) and it could just return the githash, and then also be used by the commit queue function https://github.com/ablack12/evergreen/blob/566384cbfed96d6b28eaa6678d3c087de227e625/units/commit_queue.go#L703